From a08d5d989977d57c81e090eff88c8e2260963a3f Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Mon, 11 Jun 2018 10:20:23 +0300
Subject: [PATCH 61/71] cl-som-imx8: dts: Refactoring reserved memory

1) Reduce the CMA size down to 512M.
2) Enable imx-ion-pool.

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/cl-som-imx8.dts | 76 ++++++++++++++++++++++++++--
 1 file changed, 71 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/cl-som-imx8.dts b/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
index fd785f9..fcf6003 100644
--- a/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
+++ b/arch/arm64/boot/dts/compulab/cl-som-imx8.dts
@@ -14,9 +14,6 @@
 
 /dts-v1/;
 
-/* First 128KB is for PSCI ATF. */
-/memreserve/ 0x40000000 0x00020000;
-
 #include "../freescale/fsl-imx8mq.dtsi"
 
 / {
@@ -218,6 +215,19 @@
 			>;
 		};
 
+		pinctrl_i2c3: i2c3grp {
+			fsl,pins = <
+				MX8MQ_IOMUXC_I2C3_SCL_I2C3_SCL	0x4000007f
+				MX8MQ_IOMUXC_I2C3_SDA_I2C3_SDA	0x4000007f
+			>;
+		};
+
+		touchscreen_pins: touchscreen-pins {
+			fsl,pins = <
+				MX8MQ_IOMUXC_NAND_CE3_B_GPIO3_IO4 0x17059 /* P21.34 IRQ */
+				MX8MQ_IOMUXC_NAND_CE2_B_GPIO3_IO3 0x17059 /* P21.32 RST */
+			>;
+		};
 
 		pinctrl_pcie1: pcie1grp {
 			fsl,pins = <
@@ -566,12 +576,32 @@
 
 };
 
-&pcie0{
+&i2c3 {
+    clock-frequency = <400000>;
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_i2c3>;
+    status = "okay";
+    goodix_ts@5d {
+        compatible = "goodix,gt911";
+        reg = <0x5d>;
+
+        interrupt-parent = <&gpio3>;
+        interrupts = <3 IRQ_TYPE_EDGE_FALLING>;
+
+        pinctrl-names = "default";
+        pinctrl-0 = <&touchscreen_pins>;
+
+        irq-gpios = <&gpio3 3 GPIO_ACTIVE_HIGH>;
+        reset-gpios = <&gpio3 4 GPIO_ACTIVE_HIGH>;
+    };
+};
+
+&pcie0 {
 	reset-gpio = <&pca9555 0 GPIO_ACTIVE_LOW>;
 	status = "okay";
 };
 
-&pcie1{
+&pcie1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pcie1>;
 	clkreq-gpio = <&gpio1 2 GPIO_ACTIVE_LOW>;   /* pin#39 */
@@ -790,3 +820,39 @@
 	>;
 	dc-supply = <&reg_gpio_dvfs>;
 };
+
+&resmem {
+    linux,cma {
+        compatible = "shared-dma-pool";
+        reusable;
+        size = <0 0x20000000>;
+        alloc-ranges = <0 0x40000000 0 0x40000000>;
+        linux,cma-default;
+    };
+};
+
+&resmem {
+	display_region: imx_ion@1 {
+		compatible = "imx-ion-pool";
+		reg = <0x0 0xe6000000 0 0x18000000>;
+	};
+
+	vpu_region: imx_ion@2 {
+		compatible = "imx-ion-pool";
+		reg = <0x0 0xe4000000 0 0x02000000>;
+	};
+};
+
+&imx_ion {
+	compatible = "fsl,mxc-ion", "linux,ion";
+
+	ion-display-region {
+		compatible = "fsl,display-heap", "linux,ion-heap-unmapped";
+		memory-region = <&display_region>;
+	};
+
+	ion-vpu-region {
+		compatible = "fsl,vpu-heap", "linux,ion-heap-unmapped";
+		memory-region = <&vpu_region>;
+	};
+};
-- 
2.7.4

